assets/src/blocks/CarouselHeader/CarouselHeaderFrontend.js

```diff
@@ -7,7 +7,21 @@ import { StaticCaption } from './StaticCaption';
 
 const isRTL = document.querySelector('html').dir === 'rtl';
 
+const usePageLoaded = () => {
+  const [pageLoaded, setPageLoaded] = useState(false);
```

Comment:
Why do we need a separate hook here? Couldn't we just write this inside `CarouselHeaderFrontend` directly?

Reply:
What if we want to use it in another component?

Reply:
But also to reduce complexity. `CarouselHeaderFrontend` doesn't care about anything else but that the page was loaded. So we don't need to add the complexity of the inner workings (`useState` and `useEffect`) in that place.